### PR TITLE
add customizable distance function

### LINKED ISSUE #23: feat: add version that uses a custom distance metric

### PR DESCRIPTION
**Pull request by younesStrittmatter**
adds a "cd" version of the sampler that allows for customizable distance functions. Also added a tutorial for running the model disagreement sampler with the rnn-sindy-theorist (which was a pain before but is easy with this version)

### CODE REVIEW

File: src/autora/experimentalist/model_disagreement/__init__.py [hunk 1]
```diff
@@ -9,6 +9,133 @@
 from autora.utils.deprecation import deprecated_alias
 
 
+def score_sample_cd(
```
Comment (musslick) [inline src/autora/experimentalist/model_disagreement/__init__.py]: This is minor but I'm wondering whether the function name is could be more self-explanatory? E.g., score_sample_custom_distance?

**Review by musslick — CHANGES_REQUESTED** [review]
Just a minor thing: Could we come up with a more self-explanatory function name?

File: src/autora/experimentalist/model_disagreement/__init__.py (hunk 1 above)
Comment (younesStrittmatter) [reply src/autora/experimentalist/model_disagreement/__init__.py]: renamed it to ..._custom_distance

**Review by musslick — APPROVED** [review]
Looks great!

**Pull request merged**